scalr/resource_scalr_event_bridge_integration.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/scalr/go-scalr"
+)
+
+func resourceScalrEventBridgeIntegration() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manage the state of EventBridge integrations in Scalr. Create, update and destroy.",
+		CreateContext: resourceScalrEventBridgeIntegrationCreate,
+		ReadContext:   resourceScalrEventBridgeIntegrationRead,
+		DeleteContext: resourceEventBridgeIntegrationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description:      "Name of the EventBridge integration.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			"aws_account_id": {
+				Description:      "AWS account ID.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			"region": {
+				Description:      "AWS region.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			"event_source_name": {
+				Description: "Event source name.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"event_source_arn": {
+				Description: "ARN of the event source.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceScalrEventBridgeIntegrationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+	// Get attributes.чи щось піде не так
+	name := d.Get("name").(string)
+	awsAccountID := d.Get("aws_account_id").(string)
+	region := d.Get("region").(string)
+
+	options := scalr.EventBridgeIntegrationCreateOptions{
+		Name:         &name,
+		AWSAccountId: &awsAccountID,
+		Region:       &region,
+	}
+
+	log.Printf("[DEBUG] Create EventBridge integration: %s", name)
+	integration, err := scalrClient.EventBridgeIntegrations.Create(ctx, options)
+	if err != nil {
+		return diag.Errorf("Error creating EventBridge integration %s: %v", name, err)
+	}
+	d.SetId(integration.ID)
+
+	return resourceScalrEventBridgeIntegrationRead(ctx, d, meta)
+}
+
+func resourceScalrEventBridgeIntegrationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+	integrationID := d.Id()
+
+	log.Printf("[DEBUG] Read EventBridge integration with ID: %s", integrationID)
+	EventBridgeIntegration, err := scalrClient.EventBridgeIntegrations.Read(ctx, integrationID)
+	if err != nil {
+		log.Printf("[DEBUG] EventBridge integration %s no longer exists", integrationID)
```

Comment:
It's better to separate 'not found' from the generic read error, as in https://github.com/Scalr/terraform-provider-scalr/blob/29ef9d1896dcd52e7a083146fb1a3bb4dc76280a/scalr/resource_scalr_tag.go#L48